### ISSUE TITLE
Make __fixtures__ directory configurable

### ISSUE THREAD

**Issue by ovidiubute** [react-cosmos/react-cosmos#287]
### What's up?

The `__fixtures__` directory cannot be changed from any configuration, even if you can provide additional directories to scan for fixtures, there is no way to have fixtures inside your `components` directory and call it something else. I've seen some users confused by this behavior and I believe we should provide an option to override it with whatever they like. This could also accommodate large projects that already have their own fixture structure and would never change it just to use `react-cosmos`.

### Mkay, tell me more...

Just off the top of my head, we could have a field in `cosmos.config.js` (naming could be better):

```
{
  inlineFixturesPath: '__fixtures__' (default option, can be overridden by user config)
}
```

This is going to non trivial impact, as it's used in quite a few places:

- `react-cosmos-voyager`: [here](https://github.com/react-cosmos/react-cosmos/blob/da85709af02f4ddf5b8d9908c52678f802a9f214/packages/react-cosmos-voyager/src/index.js#L6), [here](https://github.com/react-cosmos/react-cosmos/blob/ca471cb75cd1dab556a3ea68fcf3864e62015fd1/packages/react-cosmos-voyager/src/match-fixture-path.js#L2), [here](https://github.com/react-cosmos/react-cosmos/blob/da85709af02f4ddf5b8d9908c52678f802a9f214/packages/react-cosmos-voyager/scripts/generate-use-cases.js#L11), some tests as well.
- `flatris`: [here](https://github.com/react-cosmos/react-cosmos/blob/da85709af02f4ddf5b8d9908c52678f802a9f214/examples/flatris/browserify/prepare-modules.js#L27)

**Comment by skidding** [comment]
> even if you can provide additional directories to scan for fixtures, there is no way to have fixtures inside your components directory and call it something else.

You should be able to achieve this using the `getFixturePathsForComponent` option, which is not documented:

```js
// cosmos.config.js

getFixturePathsForComponent: (componentName) => {
  return glob.sync(`src/components/custom-fixture-dir/${componentName}/**/*.js`);
}
// or, if you keep fixtures per component
getFixturePathsForComponent: (componentName) => {
  return glob.sync(`src/components/${componentName}/custom-fixture-dir/**/*.js`);
}
```

Wdyt? Shall we add some docs for this option? :) 
There [one test](https://github.com/react-cosmos/react-cosmos/blob/cc89265bacbac020c3eeb8d19bbd930b3846f5f8/packages/react-cosmos-voyager/src/__tests__/index.js#L131-L139) currently.

Note that this doesn't disable reading fixtures from `__fixtures__` dir, do you think that is a concern? I'm assuming the need is to name it differently and not to use it for something else.

**Comment by ovidiubute** [comment]
While certainly possible, I wouldn't be the one who tries to explain this workaround to a noobie developer. If we could have a flag inside `cosmos.config.js` that does just what you said, but behind the scenes, it would go a long way towards helping developers.

**Comment by ovidiubute** [comment]
Also I'm not totally sold on this workaround if it doesn't actually ignore `__fixtures__` entirely. It just doesn't feel consistent to me, but that's not such a big deal as it could be fixed later.

**Comment by skidding** [comment]
@ovidiubute you're right. I'm OK with having something like `fixturesDirName` (defaulting to`__fixtures__`). Given how most users should go down this route and both `fixturePaths` and `getFixturePathsForComponent` are meant for edge cases, I wouldn't prefix the setting with "inline" nor "relative" because it would only confuse newcomers more.

> This could also accommodate large projects that already have their own fixture structure and would never change it just to use react-cosmos.

Was this the case with the people you talked you? I'm very curios what sort of fixtures did they have and what were they using them for.

**Comment by ovidiubute** [comment]
No, that last case was just something I thought about. The people I spoke to just did not want to call the directory `__fixtures__` but still liked the idea of having the fixtures in the same place as `components`.

**Comment by skidding** [comment]
Got it. So the reason behind this config is pure aesthetics. I'm OK with that. Keeping fixtures next to components (especially since we now support both `componets/__fixtures__/Foo/blank.js` and `components/Foo/__fixtures__/blank.js` out of the fox) should cover 99% of the cases. With this in mind, having the ability to customize how your project *looks* seems reasonable.

**Comment by skidding** [comment]
> This is going to non trivial impact, as it's used in quite a few places:

Note: No need to change anything in the Flatris example as the default will remain the same.

**Comment by skidding** [comment]
Fixed in #368
